pkg/operator/controllers/cpms/cpms_controller.go

```diff
@@ -0,0 +1,100 @@
+package cpms
+
+import (
+	"context"
+
+	machinev1 "github.com/openshift/api/machine/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/base"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+const (
+	ControllerName         = "CPMSController"
+	SingletonCPMSName      = "cluster"
+	SingletonCPMSNamespace = "openshift-machine-api"
+)
+
+type Reconciler struct {
+	base.AROController
+}
+
+func NewReconciler(log *logrus.Entry, client client.Client) *Reconciler {
+	return &Reconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
+	}
+}
+
+// CPMS reconciler will disable the cluster CPMS if `aro.cpms.enabled` is false or missing.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.GetCluster(ctx)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if instance.Spec.OperatorFlags.GetSimpleBoolean(operator.CPMSEnabled) {
+		r.Log.Infof("Flag %s is true, will not deactivate CPMS", operator.CPMSEnabled)
+		return reconcile.Result{}, nil
+	}
```

Comment:
nit: might be worth a comment here stating that this controller is unique in that `enabled: true` means that CPMS sticks around, and `enabled: false` means that we will deactivate it. I think it is highly clear once you see it - but it is also distinct.

Reply:
alternatively, I see the comment in the header of this function, doing some `**false**` formatting or something may draw attention to the uniqueness.